### PR TITLE
[YANG] Add route flow counter support

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-flex_counter.yang [hunk 1]
```diff
@@ -9,6 +9,10 @@ module sonic-flex_counter {
         prefix stypes;
     }
 
+    import sonic-vrf {
```
Comment (Junchao-Mellanox) [inline src/sonic-yang-models/yang-models/sonic-flex_counter.yang:12]: remove this

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/flex_counter.json [hunk 2]
```diff
@@ -97,7 +101,78 @@
...
+                    }
+                ]
+            }
+        }
+    },
+    "FLOW_COUNTER_ROUTE_PATTERN_TABLE_WITH_INVALID_MAX_MATCH_COUNT": {
+        "sonic-vrf:sonic-vrf":{
+            "sonic-vrf:VRF": {
+                "VRF_LIST": [
+                {
```
Comment (stephenxs) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/flex_counter.json:161]: Please fix indent.

File: src/sonic-yang-models/yang-models/sonic-flex_counter.yang [hunk 3]
```diff
@@ -211,8 +215,51 @@ module sonic-flex_counter {
...
+            description "Flow counter route pattern of config_db.json";
+
+            list PATTERN_LIST {
+
+                key "vrf_name ip_prefix";
+
+                leaf vrf_name {
+                    type string {
+						length 0..16;
+					}
```
Comment (stephenxs) [inline src/sonic-yang-models/yang-models/sonic-flex_counter.yang:244]: Please fix indent
Is the length defined in HLD?

File: src/sonic-yang-models/yang-models/sonic-flex_counter.yang [hunk 4]
```diff
@@ -211,8 +215,51 @@ module sonic-flex_counter {
...
+                }
+
+                leaf ip_prefix {
+					type union {
```
Comment (stephenxs) [inline src/sonic-yang-models/yang-models/sonic-flex_counter.yang:248]: Please fix indent

File: src/sonic-yang-models/yang-models/sonic-flex_counter.yang (hunk 3 above)
Comment (Junchao-Mellanox) [reply src/sonic-yang-models/yang-models/sonic-flex_counter.yang]: No, it is a validation in sonic-utilities.

**Review by stephenxs — APPROVED** [review]

**Pull request closed without merging**